hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -105,28 +109,23 @@ object PruneDeadFields {
     )
   }
 
+  def selectKey(t: TStruct, k: IndexedSeq[String]): TStruct = t.filterSet(k.toSet)._1
```

Comment:
that takes integers -- I think this may be more efficient than using `typeAfterSelect._1` though. That stuff is begging for some attention, though -- the querier stuff is barely used and can probably be ripped out